### PR TITLE
Add union bank module

### PR DESCRIPTION
**Pull request by addempsea**

### CODE REVIEW

File: lib/union/Access.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const request = require('./common/request');
+
+exports.TokenGeneration = async(data) => {
```
Comment (akinjide) [inline lib/union/Access.js]: Let's rename to just `Token`

File: lib/union/Customer.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            path: '/union/secured/cbacustomerenquiry',
+            credentials: { ...data },
+            payload: data.payload,
+        });
+    } catch (error) {
+        return error;
+    }
+};
+
+exports.CustomerAndAccountEnquiry = async(data) => {
```
Comment (akinjide) [inline lib/union/Customer.js]: Should be `exports.CustomerAccountEnquiry`

File: test/union/Access.test.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      .reply(200, AccessMock.TokenGeneration);
+    const validate = await Access.TokenGeneration(data);
+
+    expect(JSON.parse(validate)).to.have.property("message");
+    expect(JSON.parse(validate)).to.have.property("data");
+    expect(JSON.parse(validate).data).to.have.property("access_token");
+    expect(JSON.parse(validate).data).to.have.property("token_type");
+    expect(JSON.parse(validate).data).to.have.property("refresh_token");
+    expect(JSON.parse(validate).data).to.be.an("object");
+    expect(JSON.parse(validate).data.scope).to.equal("read");
```
Comment (akinjide) [inline test/union/Access.test.js]: Parse once and assert

File: lib/union/Access.js (hunk 1 above)
Comment (addempsea) [reply lib/union/Access.js]: Renamed

File: lib/union/Customer.js (hunk 2 above)
Comment (addempsea) [reply lib/union/Customer.js]: Changed.

File: test/union/Access.test.js (hunk 3 above)
Comment (addempsea) [reply test/union/Access.test.js]: Done.

File: test/union/Access.test.js [hunk 4]
```diff
@@ -10,17 +10,16 @@ const scope = nock(data.host);
 
 describe("Access", () => {
   it("Should return Token", async () => {
-    scope
-      .post("/union/oauth/token", data.payload)
-      .reply(200, AccessMock.TokenGeneration);
-    const validate = await Access.TokenGeneration(data);
+    scope.post("/union/oauth/token", data.payload).reply(200, AccessMock.Token);
+    let validate = await Access.Token(data);
+    validate = JSON.parse(validate);
```
Comment (akinjide) [inline test/union/Access.test.js]:
```javascript
 scope
    .post("/union/oauth/token", data.payload)
    .reply(200, AccessMock.Token);

let validate = JSON.parse(await Access.Token(data));
```

File: test/union/Customer.test.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    scope
+      .post("/union/secured/changeusercredentials")
+      .query(params)
+      .reply(200, CustomerMock.ChangeUserCredentials);
+    const validate = await Customer.ChangeUserCredentials({
+      sandbox_key,
+      payload: data.ChangeUserCredentials,
+      host,
+      params
+    });
```
Comment (akinjide) [inline test/union/Customer.test.js]: Inconsistent indentation

**Pull request merged**